social/src/Components/SuggestCard/SuggestCard.jsx

```diff
@@ -0,0 +1,127 @@
+import { Typography, Button, Box } from '@mui/material';
+import LoadingButton from '@mui/lab/LoadingButton';
+import React, { useEffect, useState } from 'react';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+import './SuggestCard.css';
+import '../ProfileConnect/ProfileConnect.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { setOtherUser } from '../../Features';
+import { useNavigate } from 'react-router-dom';
+import { FollowUser, UnfollowUser } from '../../Services';
+import { GetUsername } from '../../Utilities';
+import { AllSuggestions } from '..';
+
+const SuggestCard = () => {
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+    const [showSuggestions, setShowSuggestions] = useState(false);
+    const [search, setSearch] = useState('');
+    const { theme } = useSelector(state => state.theme)
+    const { allUser, token, user, loading } = useSelector(state => state.auth);
+    const Data = allUser.filter(person => person.userId!==token);
+    const Users = Data?.filter(person => !user?.following?.some(data=>data?.userId===person?.userId))
+    const TrimUsers = Users.length > 4 ? Users.slice(0, 4) : Users;
+    const [debounceText, setDebounceText] = useState('');
+    const [filteredData, setFilteredData] = useState([]);
+
+    const updateDebounceSearch = debounce(text => {
+      setDebounceText(text); 
+    }, 1000)
+
+    const handleChange = (e) => {
+      setSearch(e.target.value)
+      updateDebounceSearch(e.target.value)
+    }
+
+    function debounce (cb, delay) {
+      let timeout;
+      return (...args) => {
+        clearTimeout(timeout);
+        timeout = setTimeout(()=>{
+          cb(...args)
+        }, delay)
+      }
+    }
```

Comment:
This is something new to learn out!! Could you attach some resource about its working @nishasen ?

Reply:
Yes sure will send you